api/caseshare/real-api.ts

```diff
@@ -0,0 +1,160 @@
+import { SharedCase } from '@hmcts/rpx-xui-common-lib/lib/models/case-share.model'
+import { Request, Response } from 'express'
+import { handleGet, handlePost } from '../common/crudService'
+import { getConfigValue } from '../configuration'
+import { CASE_SHARE_PERMISSIONS, SERVICES_MCA_PROXY_API_PATH, SERVICES_RD_PROFESSIONAL_API_PATH } from '../configuration/references'
+import * as log4jui from '../lib/log4jui'
+import { JUILogger } from '../lib/models'
+import { toCaseAssigneeMappingModel } from './dtos/case-user-dto'
+import { ccdToUserDetails, prdToUserDetails } from './dtos/user-dto'
+import { CaseAssigneeMappingModel } from './models/case-assignee-mapping.model'
+import { CCDRawCaseUserModel } from './models/ccd-raw-case-user.model'
+
+const logger: JUILogger = log4jui.getLogger('real-api')
+const prdUrl: string = getConfigValue(SERVICES_RD_PROFESSIONAL_API_PATH)
+const ccdUrl: string = getConfigValue(SERVICES_MCA_PROXY_API_PATH)
+
+export async function getUsers(req: Request, res: Response) {
+  try {
+    const path = `${prdUrl}/refdata/external/v1/organisations/users?returnRoles=true&status=active`
+    const {status, data}: {status: number, data: any} = await handleGet(path, req)
+    const permissions = CASE_SHARE_PERMISSIONS.split(',')
+    const users = [...data.users]
```

Comment:
It's worth breaking out this code into a separate function ie. function doSomethingWithUsers(), so that it's easier for other devs to understand what is happening, in this filter map.

Reply:
<-- Phil to look at .